core/src/main/java/org/stellar/anchor/sep24/Sep24Service.java

```diff
@@ -266,11 +272,23 @@ public InteractiveTransactionResponse deposit(Sep10Jwt token, Map<String, String
     }
 
     if (!destinationAccount.equals(token.getAccount())) {
-      infoF(
-          "The request account:{} does not match the one in the token:{}",
-          destinationAccount,
-          token.getAccount());
-      throw new SepValidationException("'account' does not match the one in the token");
+      ClientsConfig.ClientConfig clientConfig =
+          ConfigHelper.getClientConfig(clientsConfig, token.getClientDomain(), token.getAccount());
+      if (clientConfig != null && clientConfig.getDestinationAccounts() != null) {
+        if (!clientConfig.getDestinationAccounts().contains(destinationAccount)) {
+          infoF(
+              "The request account:{} for wallet:{} is not in the allowed destination accounts list",
+              destinationAccount,
+              clientConfig.getName());
+          throw new SepValidationException("Provided 'account' is not allowed");
+        }
+      } else {
+        infoF(
+            "The request account:{} does not match the one in the token:{}",
+            destinationAccount,
+            token.getAccount());
+        throw new SepValidationException(ERR_TOKEN_ACCOUNT_MISMATCH);
+      }
```

Comment:
If I'm understanding this right, this logic will throw an error if a custodial wallet passes an account that isn't in the destination accounts list, even if that list is empty, is that correct?

My concern is that we're not supporting the case where the anchor doesn't care what the destination account is as long as the authenticated account is a trusted custodial wallet. I think we need to support this case.

Reply:
I think if the list is empty, we can assume the destination account is permitted. I don't think that complicates the implementation much

Reply:
You mean any destination account is permitted?

Reply:
It is there by default. They just need to omit list in config (it's optional)
So if they don't pass list in the config, it works as before (only auth account is allowed)

Reply:
> So if they don't pass list in the config, it works as before (only auth account is allowed)

But I'm saying thats not what we want. We should allow any destination account if the list is omitted (or empty?) and the authenticated account belongs to a custodial wallet.

Reply:
But is there a use case for that? It's very easy to add this functionality, but I'm curious why exactly we would want to do it. 
I'm concerned that adding empty list by mistake would allow arbitrary account to make deposits (bypassing their KYC policy, etc)

Reply:
Well I think its a matter of the on/off ramp's security policy and whether or not they can realistically keep these allowlists accurately over time. Its also something the custodial wallets probably would prefer.

If a custodial wallet (or exchange) updates their hot wallets frequently, or has a large number of them, I think its unreasonable to assume they will notify their on/off ramp partners of changes or that those on/off ramps will be able to update their list in a timely fashion.

Or lets say a hot wallet is compromised (or they suspect it is) so they transfer funds to another address. If their on/off ramp partners require that new address to be added to their allowlist, the wallet's on/off ramp services are effectively down.

Reply:
Let's just add a flag "allowAll" instead of using empty list for this.
So in config, they either:
1. Don't do anything (allowAll = false, list = null) -> old behavior  (only auth address is allowed) 
2. Set allowAll to true (and list will be just ignored) -> allow deposit from all sources
3. Set list to any value -> only addresses from the list are allowed
WDYT?

Reply:
Yea I think thats good too. What would the schema look like?

Reply:
The config schema? 
```yaml
     - name:
       type: custodial
       signing_key:
       callback_url:
       allow_any_destination:
       destination_accounts:
```
Something like this

Reply:
Ok -- that seems ok. I think the default should be `true` but the important piece is that the feature is available.

Last thing, how do you specify the destination accounts list in environment variables? Are these equivalent?

```
destination_accounts:
  - G1
  - G2
```

```
destination_accounts: ["G1", "G2"]
```

```
DESTINATION_ACCOUNTS=G1,G2
```

Reply:
They are not. It will be 
```
clients[1].destination_accounts[0]=G1
clients[1].destination_accounts[0]=G2
```
I don't mind to make config to be string and parse it later
So it will be
```yaml
destination_accounts = "G1, G2"
```
and env
```
DESTINATION_ACCOUNTS=G1,G2
```

Reply:
Any way we could allow the proper list form in the yaml file but a comma-separated list in the env vars? If not, I'm not sure which variation is better.

Reply:
Not in the current implementation (we need to add support for this to our configuration handlers)

Reply:
I think we should just go with lists for now. MGI is using .yaml files, and I don't think anybody else is interested in this feature at the moment.
We will re-visit client and list configurations in v3